Avoid many -Wparentheses warnings from gtk/gtk.h with gcc8

vadz opened the pull request:
Add wx/gtk/private/wrapgtk.h wrapping gtk/gtk.h in pragmas disabling
these warnings and include it everywhere instead of directly including
gtk/gtk.h.

Also include wx/gtk/private/gtk2-compat.h from this wrapper header as it
was included by 90% of the files including gtk/gtk.h itself and it seems
to be better and simpler to just always include it.

---

With this change, we avoid tons of warnings when building wx with gcc 8 as each inclusion of `<gtk/gtk.h>` resulted in

```
In file included from /usr/include/gtk-3.0/gtk/gtk.h:100,
                 from ~wx/include/wx/gtk/private/addremovectrl.h:17,
                 from ~wx/include/wx/private/addremovectrl.h:149,
                 from ~wx/src/common/addremovectrl.cpp:32:
/usr/include/gtk-3.0/gtk/gtkfilechooserbutton.h:66:8: error: unnecessary parentheses in declaration of ‘__gtk_reserved1’ [-Wparentheses]
   void (*__gtk_reserved1);
        ^
/usr/include/gtk-3.0/gtk/gtkfilechooserbutton.h:67:8: error: unnecessary parentheses in declaration of ‘__gtk_reserved2’ [-Wparentheses]
   void (*__gtk_reserved2);
        ^
/usr/include/gtk-3.0/gtk/gtkfilechooserbutton.h:68:8: error: unnecessary parentheses in declaration of ‘__gtk_reserved3’ [-Wparentheses]
   void (*__gtk_reserved3);
        ^
/usr/include/gtk-3.0/gtk/gtkfilechooserbutton.h:69:8: error: unnecessary parentheses in declaration of ‘__gtk_reserved4’ [-Wparentheses]
   void (*__gtk_reserved4);
        ^
In file included from /usr/include/gtk-3.0/gtk/gtk.h:271,
                 from ~wx/include/wx/gtk/private/addremovectrl.h:17,
                 from ~wx/include/wx/private/addremovectrl.h:149,
                 from ~wx/src/common/addremovectrl.cpp:32:
/usr/include/gtk-3.0/gtk/deprecated/gtkstatusicon.h:74:8: error: unnecessary parentheses in declaration of ‘__gtk_reserved1’ [-Wparentheses]
   void (*__gtk_reserved1);
        ^
/usr/include/gtk-3.0/gtk/deprecated/gtkstatusicon.h:75:8: error: unnecessary parentheses in declaration of ‘__gtk_reserved2’ [-Wparentheses]
   void (*__gtk_reserved2);
        ^
/usr/include/gtk-3.0/gtk/deprecated/gtkstatusicon.h:76:8: error: unnecessary parentheses in declaration of ‘__gtk_reserved3’ [-Wparentheses]
   void (*__gtk_reserved3);
        ^
/usr/include/gtk-3.0/gtk/deprecated/gtkstatusicon.h:77:8: error: unnecessary parentheses in declaration of ‘__gtk_reserved4’ [-Wparentheses]
   void (*__gtk_reserved4);
        ^
```

vadz committed:
Avoid many -Wparentheses warnings from gtk/gtk.h with gcc8

Add wx/gtk/private/wrapgtk.h wrapping gtk/gtk.h in pragmas disabling
these warnings and include it everywhere instead of directly including
gtk/gtk.h.

Also include wx/gtk/private/gtk2-compat.h from this wrapper header as it
was included by 90% of the files including gtk/gtk.h itself and it seems
to be better and simpler to just always include it.

paulcor commented on src/gtk/window.cpp:
```diff
@@ -37,16 +37,15 @@
 #include "wx/private/textmeasure.h"
 using namespace wxGTKImpl;
 
+#include <gdk/gdkkeysyms.h>
```
This will cause a compile error with GTK+ < 2.20. `gdkkeysyms.h` needs to come before `gtk2-compat.h`.

vadz replied:
Are you sure about this? AFAICS we don't really use the constants from `gdkkeysyms.h` in `wx/gtk/private/gtk2-compat.h`, we just do `#define GDK_KEY_XXX GDK_XXX`. And it only matters that `GDK_XXX` are defined before they're used in `src/gtk/window.cpp`.

Or am I missing something here?

paulcor replied:
```
../src/gtk/window.cpp: In function ‘long int wxTranslateKeySymToWXKey(KeySym, bool)’:
../src/gtk/window.cpp:564:14: error: ‘GDK_KEY_Shift_L’ was not declared in this scope
         case GDK_KEY_Shift_L:
              ^~~~~~~~~~~~~~~
```
`gtk2-compat.h` won't define the new names unless the old names are already defined (i.e. `gdkkeysyms.h` has already been included.

vadz replied:
> `gtk2-compat.h` won't define the new names unless the old names are already defined

Ah, this was what I was missing, thanks.

The simplest way to fix this would be to just always include `gdk/gdkkeysyms.h` from `gtk2-compat.h`. Do you see any problems with this? Looking at the GDK header, it's a pretty trivial one, so there shouldn't be any risk nor overhead from including it from there.

paulcor reviewed (commented):
I'm a little conflicted on this, not sure it's worth the effort. The problem has already been fixed in the GTK+ headers, so in the future this will just be for building with "old" versions of GTK+.

vadz commented:
I'd prefer to commit it because it bothers me right now and because I think it might be convenient to have another central place for common GTK+-related declarations in the future (e.g. maybe we're going to move `wx/gtk/private/gtk3-compat.h` inclusion into it later too), so if you don't object, I'm going to do it.

Please let me know if you do.

vadz committed:
Fix build with GTK+ < 2.20 due to undefined key code constants

Include gdk/gdkkeysyms.h from wx/gtk/private/gtk2-compat.h as it must be
included before it for the compatibility header to define the new names
for the key constants even when using old GTK+ versions.
